crates/rolldown/src/ecmascript/format/iife.rs

```diff
@@ -65,14 +78,79 @@ pub fn render_iife(
   }
 
   // iife wrapper end
-  concat_source.add_source(Box::new(RawSource::new(format!(
-    "}})({});",
-    if has_exports && named_exports { "{}" } else { "" }
-  ))));
+  if invoke {
+    concat_source.add_source(Box::new(RawSource::new(format!("}})({output_args});"))));
+  } else {
+    concat_source.add_source(Box::new(RawSource::new("})".to_string())));
+  }
 
   if let Some(footer) = footer {
     concat_source.add_source(Box::new(RawSource::new(footer)));
   }
 
   Ok(concat_source)
 }
+
+// Handling external imports needs to modify the arguments of the wrapper function.
+fn render_iife_chunk_imports(ctx: &GenerateContext<'_>) -> (String, Vec<String>) {
+  let render_import_stmts =
+    collect_render_chunk_imports(ctx.chunk, ctx.link_output, ctx.chunk_graph);
+
+  let mut s = String::new();
+  let externals: Vec<String> = render_import_stmts
+    .iter()
+    .filter_map(|stmt| {
+      let require_path_str = format!("{}", &stmt.path);
+      match &stmt.specifiers {
+        RenderImportDeclarationSpecifier::ImportSpecifier(specifiers) => {
+          // Empty specifiers can be ignored in IIFE.
+          if specifiers.is_empty() {
+            None
+          } else {
+            let specifiers = specifiers
+              .iter()
+              .map(|specifier| {
+                if let Some(alias) = &specifier.alias {
+                  format!("{}: {alias}", specifier.imported)
+                } else {
+                  specifier.imported.to_string()
+                }
+              })
+              .collect::<Vec<_>>();
+            s.push_str(&format!(
+              "const {{ {} }} = {};\n",
+              specifiers.join(", "),
+              legitimize_identifier_name(require_path_str.as_str())
+            ));
+            Some(require_path_str)
+          }
+        }
+        RenderImportDeclarationSpecifier::ImportStarSpecifier(alias) => {
+          s.push_str(&format!("const {alias} = {require_path_str};\n"));
```

Comment:
Here also need to using  legitimize_identifier_name

Reply:
Solved. I performed `legitimize_identifier_name` in assigning to `require_path_str`.

```rust
      let require_path_str = legitimize_identifier_name(&stmt.path).to_string();
```